modules/src/food.py

```diff
@@ -0,0 +1,93 @@
+import os
+import json
+import requests
+import random
+import difflib
+
+import config
+from templates.generic import *
+from templates.text import TextTemplate
+
+ZOMATO_API_KEY = os.environ.get('ZOMATO_API_KEY', config.ZOMATO_API_KEY)
+
+def process(input,entities):
+    output = {}
+    try:
+        requested_location = entities['restaurant_location'][0]['value']
+
+        header = {'user-key': ZOMATO_API_KEY, 'Accept': 'application/json'}
+        query = requests.get('https://developers.zomato.com/api/v2.1/locations?query=' + requested_location, headers=header).json()
+        
+        location = query['location_suggestions'][0]
+
+        search_url = 'https://developers.zomato.com/api/v2.1/search?entity_id={eid}&entity_type={etype}&lat={lat}&lon={lon}&sort=rating&order=desc'.format(eid=location['entity_id'],etype=location['entity_type'],lat=location['latitude'],lon=location['longitude'])
+        
+        name = ''
+        type = ''
+
+        if 'restaurant_name' in entities:
+            name = entities['restaurant_name'][0]['value']
+            search_url += '&q={name}'.format(name=name)
+
+        if 'cuisine' in entities:
+            type = entities['cuisine'][0]['value']
+            search_url += '&cuisines={type}'.format(type=type)
+
+        query = requests.get(search_url,headers=header).json()['restaurants']
+
+        if name == '':
```

Comment:
Could you please explain what is happening from here :) a bit confusing and may be I sense that it could be simplified from the line 25, And it would be better if we have a separate function with the readable name when the code is longer.

Reply:
Basically, queries the api again for more specific results on a specific restaurant or type of food from the input